File: Easy/167_TwoSumII-InputArrayIsSorted.php
<?php
class Solution {

    /**
     * @param Integer[] $numbers
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($numbers, $target) {
        $i = 0;
        while($i < (count($numbers))) {
            $a = $target - $numbers[$i];
            $b = array_search($a, array_slice($numbers, ($i + 1)));
            if (gettype($b) == 'integer') {
                return [($i+1), ($b + $i + 2)];
            }
            $i++;
        }        
        return [];        
    }
}

File: Medium/167_TwoSumII-InputArrayIsSorted.rb
# @param {Integer[]} numbers
# @param {Integer} target
# @return {Integer[]}
def two_sum(numbers, target)
    left = 0
    right = numbers.length - 1

    while left < right
        sum = numbers[left] + numbers[right]

        if sum == target
        return [left + 1, right + 1]  # Convert to 1-based indexing
        elsif sum < target
        left += 1
        else
        right -= 1
        end
    end
end
